src/main/src/main/java/org/geoserver/config/SpringResourceAdaptor.java

```diff
@@ -74,22 +86,30 @@ public boolean isOpen() {
 
     @Override
     public URL getURL() throws IOException {
-        return Resources.find(resource).toURI().toURL();
+        return getFile().toURI().toURL();
     }
 
     @Override
     public URI getURI() throws IOException {
-        return Resources.find(resource).toURI();
+        return getFile().toURI();
     }
 
+    /**
+     * @return a File handle for this resource.
+     * @throws FileNotFoundException if the resource is UNDEFINED.
+     */
     @Override
     public File getFile() throws IOException {
-        return Resources.find(resource);
+        File file = Resources.find(resource);
+        if (file == null) {
+            throw new FileNotFoundException(resource.path());
+        }
+        return file;
     }
 
     @Override
     public long contentLength() throws IOException {
-        return Resources.find(resource).length();
+        return getFile().length();
```

Comment:
It is these kind of method where I want to check the javadocs; and see if we could return ``0`` or ``-1`` or something to be kinder to client code. Still this is a judgement call and you are the person doing the work so I trust your judgement :)

Reply:
yes, I see. I was too fast to replace them all. I will check again.